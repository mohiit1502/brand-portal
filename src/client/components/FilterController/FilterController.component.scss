.c-FilterController {
  height: 16.67%;
}
