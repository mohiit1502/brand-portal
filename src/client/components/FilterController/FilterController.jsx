import React, {memo, useState} from "react";
import PropTypes from "prop-types";
import "./FilterController.component.scss";
import * as images from "../../images";

const FilterController = props => {
  const [containerState, setContainerState] = useState({
    className: "form-group custom-input-form-group custom-select-form-group dropdown"
  });
  const [fieldState, setFieldState] = useState({
    className: `form-control form-control-filter-controller custom-input-element`,
    dropdownOptions: {
      weekWise: [
        {id: "yesterday", value: "yesterday", label: "Yesterday"},
        {id: "lastweek", value: "lastweek", label: "Last Week"},
        {id: "weektodate", value: "weektodate", label: "Week to Date"}
      ],
      monthWise: [
        {id: "monthtodate", value: "monthtodate", label: "Month to Date"},
        {id: "lastmonth", value: "lastmonth", label: "Last Month"}
      ],
      quarterWise: [
        {id: "quartertodate", value: "quartertodate", label: "Quarter to Date"},
        {id: "lastquarter", value: "lastquarter", label: "Last Quarter"}
      ],
      yearWise: [
        {id: "yeartodate", value: "yeartodate", label: "Year to Date"},
        {id: "lastyear", value: "lastyear", label: "Last Year"},
        {id: "alltime", value: "alltime", label: "All time"}
      ],
      custom: [
        {id: "customdate", value: "customdate", label: "Custom Date", handler: dispatchDateSelector}
      ]
    },
    error: "",
    id: "claim-type-widget-filter-controller",
    subTitle: "",
    type: "select",
    value: ""
  })

  const dispatchDateSelector = () => {

  }

  const updateChart = () => {

  }

  return (
    <div className="c-FilterController">
      <div className={containerState.className}>
        <input type={fieldState.type} className={fieldState.className} id={fieldState.id} value={fieldState.value ? fieldState.value : "All Time"} onChange={updateChart}
               data-toggle="dropdown" autoComplete="off" />
        <label className="custom-input-label custom-input-label-placeholder" htmlFor="claim-type-widget-filter-controller">
          {/*<div className="label-upper-bg position-absolute w-100 h-50 d-block"/>*/}
          <div className="label-lower-bg position-absolute w-100 h-50 d-block"/>
          <span className="label-text"> Time range </span>
        </label>
        <img src={images.ArrowDown} alt="image-arrow-down" className="dropdown-arrow"/>
        <div className="dropdown-menu">
          {Object.keys(fieldState.dropdownOptions).map(ddSection => {
            const sectionArray = fieldState.dropdownOptions[ddSection];
            return <div className="border-bottom">
              {sectionArray.map((option, i) => <a key={option.id || i} className="dropdown-item" onClick={option.handler ? option.handler : () => setFieldState({...fieldState, value: option.label})}>{option.label}</a>)}
            </div>
          })}
        </div>
      </div>
    </div>
  );
};

FilterController.propTypes = {

};

export default memo(FilterController);
