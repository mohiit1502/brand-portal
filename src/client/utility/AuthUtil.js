import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
