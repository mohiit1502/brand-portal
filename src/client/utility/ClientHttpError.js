export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
