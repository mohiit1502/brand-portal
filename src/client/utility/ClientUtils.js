import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
