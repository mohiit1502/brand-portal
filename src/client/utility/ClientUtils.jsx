import React from "react";
import queryString from "query-string";
import {Redirect} from "react-router";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }

  static  getAllValuesFromRecursiveTree(tree) {

    if (typeof tree !== "object" || tree === null) {
      return [tree];
    }

    const list = [];

    for (const i in tree) {
      if (tree.hasOwnProperty(i)) {
        list.push(...this.getAllValuesFromRecursiveTree(tree[i]));
      }
    }
    return list;
  }

  static where (arrayList, filterObj) {
    let index = -1;
    for (let i = 0; i < arrayList.length; i++) {
      for (const j in filterObj) {
        if (filterObj.hasOwnProperty(j) && arrayList[i].hasOwnProperty(j) && filterObj[j] === arrayList[i][j]) {
          index = i;
        } else {
          index = -1;
        }
      }
      if (index !== -1) {
        return index;
      }
    }
    return index;
  }

}
