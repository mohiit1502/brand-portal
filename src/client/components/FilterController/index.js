import FilterController from './FilterController.jsx';

export default FilterController;