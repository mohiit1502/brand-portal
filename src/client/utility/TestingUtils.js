import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
