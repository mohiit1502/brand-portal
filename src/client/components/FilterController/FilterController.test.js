import React, {useRef} from "react";
import FilterController from "./FilterController";
import toJson from "enzyme-to-json";
import {mount} from "enzyme";
import {clearKeys, testStore} from "../../utility/TestingUtils";
import {Provider} from "react-redux";
import MockNextContext from "../../../../test/client/utility/MockNextContext";


let props;
let store;

const setUp = (props) => {
  store = testStore({
    dashboard:{
      customDate:{}
    }
  });
  return mount(props ? <Provider store={store}><MockNextContext pathname="/help"><FilterController {...props} /></MockNextContext></Provider>: <Provider store={store}><MockNextContext pathname="/help"><FilterController /></MockNextContext></Provider>);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});



describe("FilterController", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });
  it("renders without error", () => {
    props = {
      customDate: {},
      currentFilters: {},
      dispatchFilter: () => {},
      filters: [],
      toggleModal: () => {},
      updateChartMeta: {},
      widgetId: ""
    };
    wrapper = setUp(props);

    const tree = toJson(wrapper);
    clearKeys(tree,[]);
    expect(tree).toMatchSnapshot();

  });
});
