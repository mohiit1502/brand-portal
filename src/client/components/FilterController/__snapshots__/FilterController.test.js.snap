// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterController renders without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MockNextContext
    pathname="/help"
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/help",
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/help",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/help",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(FilterController)
          currentFilters={Object {}}
          customDate={Object {}}
          dispatchFilter={[Function]}
          filters={Array []}
          toggleModal={[Function]}
          updateChartMeta={Object {}}
          widgetId=""
        >
          <FilterController
            currentFilters={Object {}}
            customDate={Object {}}
            dispatchFilter={[Function]}
            filters={Array []}
            toggleModal={[Function]}
            updateChartMeta={Object {}}
            widgetId=""
          >
            <div
              className="c-FilterController row"
            />
          </FilterController>
        </Connect(FilterController)>
      </Router>
    </MemoryRouter>
  </MockNextContext>
</Provider>
`;
